async.js

```diff
@@ -0,0 +1,33 @@
+"use strict"
+
+/* globals Promise:true */
+
+function makeAsyncApi(api) {
+    return Object.keys(api).reduce(function (asyncApi, method) {
+        if (isWrapableMethod(method, api)) {
+            asyncApi[method] = createAsyncMethodProxy(api[method])
+        }
+
+        return asyncApi
+    }, {})
+}
+
+function isWrapableMethod(method, api) {
+    return method !== "async" &&
+        method !== "AssertionError" &&
+        typeof api[method] === "function"
+}
+
+function createAsyncMethodProxy(method) {
+    return function proxy(expr) {
+        var rest = [].slice.call(arguments, 1)
+
+        return Promise.resolve(expr)
+            .then(function (value) {
+                method.apply(undefined, [value].concat(rest))
+            }
+        )
+    }
+}
+
+module.exports = makeAsyncApi
```

Comment:
But this can lead to errors 
when someone adds a new assertion type and forgets to add it to async submodule